docs/developer-notes/ARCHITECTURE.md

```diff
@@ -0,0 +1,334 @@
+    ...props,
+    description: 'Your service description',
+  });
+}
+```
+
+#### Service Stack Constructor Pattern
+
+All services must follow this standardized constructor signature:
+
+```typescript
+export class YourServiceStack extends Stack {
+  constructor(
+    scope: Construct,
+    id: typeof SERVICE_NAME | (string & {}),
+    props: YourServiceStackProps
+  ) {
+    super(scope, id, props);
+    // Implementation
+  }
+}
+```
+
+### 2. Infrastructure and Runtime Code Separation
+
+Maintain strict separation between infrastructure and runtime code:
+
+```
+services/[service-name]/
+├── src/
+│   ├── index.ts              # Main stack definition
+│   ├── service-name.ts       # Service name constant
+│   ├── infra/                # Infrastructure-only code
+│   │   ├── functions/        # Lambda construct definitions
+│   │   ├── buckets/          # S3 bucket constructs
+│   │   ├── step-functions/   # Step Function definitions (YAML)
+│   │   └── environment/      # Environment parameters
+│   └── runtime/              # Runtime-only code
+│       ├── handlers/         # Lambda handler implementations
+│       └── lib/              # Shared runtime utilities
+└── tests/                    # Test files
+```
+
+#### Key Separation Rules:
+
+- **Infra code**: CDK constructs, resource definitions, configuration
+- **Runtime code**: Lambda handlers, business logic, utilities
+- **No mixing**: Runtime code cannot import from infra, and vice versa
+- **Asset resolution**: Use `resolveAssetPath()` to reference runtime assets from infra
+
+### 3. Default Constructs and Utilities
+
+#### LambdaFunction Construct
+
+Use the standardized `LambdaFunction` construct from `@libs/cdk-utils`:
+
+```typescript
+import { LambdaFunction } from '@libs/cdk-utils/infra';
+
+const myFunction = new LambdaFunction(this, 'MyFunction', {
+  entry: resolveAssetPath('runtime/handlers/my-handler.ts'),
+  environment: {
+    DATA_BUCKET: bucket.bucketName,
+  },
+});
+```
+
+**Benefits:**
+
+- Stage-aware bundling optimizations
+- Automatic log group creation with appropriate retention
+- Environment variable management
+- Source map support for debugging
+
+#### Step Functions
+
+Use `StepFunctionFromFile` for YAML-based definitions:
+
+```typescript
+import { StepFunctionFromFile } from '@libs/cdk-utils/infra';
+
+const stepFunction = new StepFunctionFromFile(this, 'ProcessWorkflow', {
+  filepath: resolveAssetPath('infra/step-functions/process-workflow.asl.yaml'),
+  lambdaFunctions: [functionA, functionB, functionC], // Automatic ARN resolution
+});
+```
+
+#### S3 Buckets
+
+Use specialized bucket constructs:
+
+```typescript
+import { TemporaryDataBucket, ConfigBucket } from '@libs/cdk-utils/infra';
+
+// For short-lived data with automatic lifecycle policies
+const dataBucket = new TemporaryDataBucket(this, 'DataBucket');
+
+// For configuration and long-term storage
+const configBucket = new ConfigBucket(this, 'ConfigBucket');
+```
+
+#### SSM Parameters
```

Comment:
Is this the appropriate spot to include something about using the CSV and JSON files to define values for these?